File: 2003/CS101SamplesAll/CS101SamplesVSTO/UsingTheListObject/UsingTheListObject/AddProductUserControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

using Office = Microsoft.Office.Core;
using Excel = Microsoft.Office.Interop.Excel;

namespace UsingTheListObject
{
    public partial class AddProductUserControl : UserControl
    {
        AdventureWorks_DataDataSet dataSet;
        AdventureWorks_DataDataSetTableAdapters.ProductTableAdapter tableAdapter;
        AdventureWorks_DataDataSet.ProductDataTable productTable;
        AdventureWorks_DataDataSet.ProductRow currentProduct = null;

        public AddProductUserControl()
        {
            InitializeComponent();

            dataSet = new AdventureWorks_DataDataSet();
            productTable = dataSet.Product;

            tableAdapter = new UsingTheListObject.AdventureWorks_DataDataSetTableAdapters.ProductTableAdapter();
            tableAdapter.Fill(productTable);
        }

        /// <summary>
        /// The Insert button should only be enabled if the item
        /// name has been found.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void itemNumberTextbox_TextChanged(object sender, EventArgs e)
        {
            if (itemNumberTextBox.Text.Length == 0)
            {
                lookupButton.Enabled = false;
                currentProduct = null;
                insertItemButton.Enabled = false;
            }
            else
            {
                lookupButton.Enabled = true;
            }
        }

        private void lookupButton_Click(object sender, EventArgs e)
        {
            AdventureWorks_DataDataSet.ProductRow[] row =
                (AdventureWorks_DataDataSet.ProductRow[])
                dataSet.Product.Select("ProductNumber='" + itemNumberTextBox.Text + "'");

            if (row.Length > 0)
            {
                itemNameTextbox.Text = row[0].Name;
                currentProduct = row[0];
            }
            else
            {
                itemNameTextbox.Text = "{NOT FOUND}";
                currentProduct = null;
            }

            insertItemButton.Enabled = (currentProduct != null);
        }

        private void insertItemButton_Click(object sender, EventArgs e)
        {
            Excel.ListRow row = Globals.Sheet1.orderDetailsList.ListRows.Add(Type.Missing);
            Excel.Range currentCell = row.Range.get_Item(1, 1) as Excel.Range;
            currentCell.Value2 = currentProduct.ProductNumber;

            currentCell = row.Range.get_Item(1, 2) as Excel.Range;
            currentCell.Value2 = currentProduct.Name;

            currentCell = row.Range.get_Item(1, 3) as Excel.Range;
            currentCell.Value2 = currentProduct.ListPrice;
        }
    }
}

File: 2003/VB101SamplesAll/VB101SamplesVSTO/UsingTheListObject/UsingTheListObject/AddProductUserControl.vb
Imports Office = Microsoft.Office.Core
Imports Excel = Microsoft.Office.Interop.Excel

Public Class AddProductUserControl
    Dim dataSet As AdventureWorks_DataDataSet
    Dim tableAdapter As AdventureWorks_DataDataSetTableAdapters.ProductTableAdapter
    Dim productTable As AdventureWorks_DataDataSet.ProductDataTable
    Dim currentProduct As AdventureWorks_DataDataSet.ProductRow

    Public Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        dataSet = New AdventureWorks_DataDataSet()
        productTable = dataSet.Product

        tableAdapter = New UsingTheListObject.AdventureWorks_DataDataSetTableAdapters.ProductTableAdapter()
        tableAdapter.Fill(productTable)
    End Sub

    ''' <summary>
    ''' The Insert button should only be enabled if the item
    ''' name has been found.
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Sub itemNumberTextbox_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles itemNumberTextBox.TextChanged
        If (itemNumberTextBox.Text.Length = 0) Then
            lookupButton.Enabled = False
            currentProduct = Nothing
            insertItemButton.Enabled = False
        Else
            lookupButton.Enabled = True
        End If
    End Sub

    Private Sub lookupButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles lookupButton.Click
        Dim row() As AdventureWorks_DataDataSet.ProductRow = CType(dataSet.Product.Select(("ProductNumber='" _
                        + (itemNumberTextBox.Text + "'"))), AdventureWorks_DataDataSet.ProductRow())
        If (row.Length > 0) Then
            itemNameTextbox.Text = row(0).Name
            currentProduct = row(0)
        Else
            itemNameTextbox.Text = "{NOT FOUND}"
            currentProduct = Nothing
        End If
        insertItemButton.Enabled = (Not (currentProduct) Is Nothing)
    End Sub

    Private Sub insertItemButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles insertItemButton.Click
        Dim row As Excel.ListRow = Globals.Sheet1.orderDetailsList.ListRows.Add(Type.Missing)
        Dim currentCell As Excel.Range = CType(row.Range.Item(1, 1), Excel.Range)
        currentCell.Value2 = currentProduct.ProductNumber
        currentCell = CType(row.Range.Item(1, 2), Excel.Range)
        currentCell.Value2 = currentProduct.Name
        currentCell = CType(row.Range.Item(1, 3), Excel.Range)
        currentCell.Value2 = currentProduct.ListPrice
    End Sub

End Class
